### PR TITLE
[Draft] Contributing guidelines

### PR DESCRIPTION
**Pull request by FrostTheFox**
Adds contributing guidelines to be referenced when opening issues/pull requests.
## Description

Adds a CONTRIBUTING.md file.
## Motivation and Context

Solves some common questions about contributing to the project.

Completely open to edits, a first draft that I wrote running on no sleep and a bunch of caffeine.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+## I just have a question/need help.
+
+Please do not open a GitHub issue for support or questions. They are best answered via [Discord](https://discord.gg/RocketMap). GitHub issues are for RocketMap code issues *only*.
+
+## How do I get started?
+
+RocketMap has two main dependencies to get started:
+
+1. RocketMap is a Python2 project, therefor, you need [Python 2](https://www.python.org/downloads/)!
```
Comment (pogo-excalibur) [inline CONTRIBUTING.md]: therefor -> therefore
(and possibly remove the second comma?)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+2. RocketMap also requires [NodeJS](https://nodejs.org/en/download/) in order to minify frontend assets.
```
Comment (pogo-excalibur) [inline CONTRIBUTING.md]: (Suggestion) minify -> build ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+### NodeJS Packages
+
+NodeJS packages are found in package.json in the project root, and can be easily installed by simply executing `npm install`.
+
+## Reporting code issues
+
+### Before you report
+
+1. **Make sure you are on the latest version!** When in doubt, `git pull`.
+2. Ensure it is an actual issue with the project itself. Confirm that you do not have faulty config, and that you have properly installed all the required dependencies. ***READ THE ERROR OUTPUT YOU ARE GIVEN!*** It may have the solution readily printed for you!
```
Comment (pogo-excalibur) [inline CONTRIBUTING.md]: 'have faulty config' -> 'have **a** faulty config'

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+3. Reproduce. Confirm the circumstances you are able to reproduce the bug. If specific configuration is required, include in your report the configuration required. Find a friend, spin up a virtual machine, or use a different computer and test the issue there. If you do not have the issue on the new machine, it may be indicative that it is something with your configuration.
+4. *USE GITHUB SEARCH.* Search open issues and look for any open issues that already address the problem. Duplicate issues will be closed.
+
+### Submitting a helpful bug report
+
+When you begin the process of opening an issue, you will be given a template in the text box that should help guide you through the process.
+
```
Comment (pogo-excalibur) [inline CONTRIBUTING.md]: (Suggestion) Add 'Please fill in as many fields as you can.'

**Comment by FrostTheFox** [comment]
@pogo-excalibur addressed, thanks :)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+To contribute code to RocketMap, it is heavily advised that you are knowledgeable in Git. At a minimum, you should know how to commit, push, pull and do basic rebasing.
+
+### Styleguides
+
+RocketMap follows and adheres to the [pep8 standards](https://www.python.org/dev/peps/pep-0008/) for Python code, and has established rules for JavaScript in the .eslintrc.json file in the project root. Automatic checks of these standards are performed when pull requests are opened, however you can save some time by running these checks on your local machine. 
+
+To check if your Python code conforms to PEP8, you can use the flake8 package (`pip install flake8`). After making changes, open a terminal in the project root and run `flake8 --statistics --show-source --disable-noqa`.
+
+To check if your Python code conforms to the eslint rules, you can run `npm run lint`.
```
Comment (friscoMad) [inline CONTRIBUTING.md]: Js

**Pull request merged**